vivid-server/src/modules/blocks/blocks.service.ts

```diff
@@ -0,0 +1,59 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { DeleteResult, Repository, UpdateResult } from 'typeorm';
+import { BlocksEntity } from '@/blocks.entity';
+
+@Injectable()
+export class BlocksService {
+  constructor(
+    @InjectRepository(BlocksEntity)
+    private blocksRepository: Repository<BlocksEntity>,
+  ) {}
+
+   // see all blocks
+  async findAll(): Promise<BlocksEntity[]> {
+    return await this.blocksRepository.find();
+  }
+
+  // block user
+  async block(userId: string, blockId: string)
+  : Promise<UpdateResult | void> {
+    return await this.blocksRepository
+      .createQueryBuilder()
+      .insert()
+      .values(
+        {
+          user_id: userId,
+          blocked_user_id: blockId,
+        },
+      )
+      .execute()
+      .catch((error) => {
+        if (error.code === '23505') throw new BadRequestException();
```

Comment:
forgot to throw default so
```js
if (...) throw ...;
throw error;
```

Reply:
do that on all of them as well. then you remove void from return type everywhere as well.